diff --git a/src/MetatagManager.php b/src/MetatagManager.php
index 2d84bcb4..7a6b4983 100644
--- a/src/MetatagManager.php
+++ b/src/MetatagManager.php
@@ -276,20 +276,20 @@ class MetatagManager implements MetatagManagerInterface {

     // Sort the tags based on the group.
     $sorted_tags = [];
-    foreach ($this->sortedGroups() as $group_name => $group) {
-      $tag_weight = $group['weight'] * 100;
-
-      // First, sort the tags within the group according to the original sort
-      // order provided by the tag's definition.
-      uasort($tags[$group_name], [
-        'Drupal\Component\Utility\SortArray',
-        'sortByWeightElement',
-      ]);
-      foreach ($tags[$group_name] as $tag_name => $tag_info) {
-        $tag_info['weight'] = $tag_weight++;
-        $sorted_tags[$tag_name] = $tag_info;
-      }
-    }
+    // foreach ($this->sortedGroups() as $group_name => $group) {
+    //   $tag_weight = $group['weight'] * 100;
+
+    //   // First, sort the tags within the group according to the original sort
+    //   // order provided by the tag's definition.
+    //   uasort($tags[$group_name], [
+    //     'Drupal\Component\Utility\SortArray',
+    //     'sortByWeightElement',
+    //   ]);
+    //   foreach ($tags[$group_name] as $tag_name => $tag_info) {
+    //     $tag_info['weight'] = $tag_weight++;
+    //     $sorted_tags[$tag_name] = $tag_info;
+    //   }
+    // }

     return $sorted_tags;
   }
